APISpec.md

```diff
@@ -97,3 +97,247 @@ Prerequisites are formatted as follows:
 ]` Means in order to take the course, you must have completed course id `2983` AND have completed `734` OR `1673`.
 
 ## Majors
+
+#### `GET /majors/`
+
+Returns a list of all majors.
+
+#### Response:
+
+```json
+[
+  {
+    "major_id": 34,
+    "name": "Computer Science",
+    "degree": "BS"
+  },
+  {
+    "major_id": 34,
+    "name": "Comparative Ethnic Studies",
+    "degree": "BA"
+  }
+  // ...
+]
+```
+
+#### `GET /majors/{major_id}`
+
+Returns data about a specific major.
+
+#### Response:
+
+```json
+
+{
+  "major_id": 34,
+  "name": "Computer Science",
+}
+
+```
+
+## Concentrations
+
+#### `GET /concentration/`
+
+Returns a list of all concentrations.
+
+#### Response:
+
+```json
+[
+  {
+    "major_id": 34,
+    "concentration_id": 56,
+    "name": "General Curriculum in Computer Science"
+  },
+  {
+    "major_id": 34,
+    "concentration_id": 57,
+    "name": "Artificial Intelligence and Machine Learning"
+  }
+  // ...
+]
+```
+
+#### `GET /concentration/{concentration_id}`
+
+Returns data about a specific concentration.
+
+#### Response:
+
+```json
+{
+  "major_id": 34,
+  "concentration_id": 56,
+  "name": "General Curriculum in Computer Science"
+}
+```
+
+## Users
+
+#### `POST /user/`
+
+Creates a new account.
+
+#### Request:
+
+```json
+{
+  "email": "example@gmail.com",
+  "first_name": "John",
+  "last_name": "Doe"
+}
+```
+
+#### Response:
+
+```json
+{
+  "user_id": 1
+}
+```
+
+#### `POST /user/{user_id}/course_history/{course_id}`
+
+Updates completed courses to users profile.
+
+#### Request:
+
+```json
+{ 
+  "completed": "boolean"
+}
+```
+
+#### Response:
+
+```json
+{
+  "response": 200
+}
+```
+
+## Plan
+
+#### `POST /plan/`
+
+Returns a plan for a specified degree, major and concentration combo.
+
+#### Request:
+
+```json
+{ 
+  "user_id": 1,
+  "degree": "BS",
+  "major_id": 34,
+  "concentration_id": 57
+}
+```
+
+#### Response:
+
+```json
+{
+  "plan_id": 1,
+  "courses": [1155, 781, 901, ... ]
+}
+```
+
+
+#### `POST /plan/update/`
+
+Returns an updated plan for a specified degree, major and concentration combo, previous plan will be overwritten.
+
+#### Request:
+
+```json
+{ 
+  "user_id": 1,
+  "plan_id": 1,
+  "degree": "BS",
+  "major_id": 34,
+  "concentration_id": 57
+}
+```
+
+#### Response:
+
+```json
+{
+  "plan_id": 1,
+  "courses": [1155, 781, 901, ... ]
+}
+```
+
+#### `POST /plan/delete/`
+
+Returns an updated plan for a specified degree, major and concentration combo, previous plan will be overwritten.
+
+#### Request:
+
+```json
+{ 
+  "user_id": 1,
+  "plan_id": 1
+}
+```
+
+#### Response:
+
+```json
+{
+  "response": 200
+}
+```
+
+#### `DELETE /plan/{plan_id}/`
+
+Returns an updated plan for a specified degree, major and concentration combo, previous plan will be overwritten.
+
+#### Request:
+
+```json
+{ 
+  "user_id": 1,
+  "plan_id": 1
+}
+```
+
+#### Response:
+
+```json
+{
+  "response": 200
+}
+```
+
+#### `GET /plan/{user_id}/`
```

Comment:
Hmm, I'm wondering if {user_id} is necessary here, since there's not really ever a case where you'd need to request a different user's plan.

What if instead we have two routes, `GET /plan/` to return a list of all plans associated with the logged in user, and `GET /plan/{plan_id}` to return a specific plan.